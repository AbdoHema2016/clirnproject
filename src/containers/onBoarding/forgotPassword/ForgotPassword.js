import React from 'react';
import {View, KeyboardAvoidingView, Platform} from 'react-native';
import Style from './style';
import {connect} from 'react-redux';
import {forgottenPasswordAction, addEmailAction} from './redux/actions';
import CLabel from '../../../components/cLabel';
import CImage from '../../../components/cImage';
import CText from '../../../components/cTextField';
import CButton from '../../../components/cButton';
import {
  Constants,
  HelperFunctions,
  Validations,
  testIds,
} from '../../../utilities';
import {translate} from '../../../Localization';

import CNavigationBackButton from '../../../components/cNavigationBackButton';

const {LOCAL_PATH} = Constants;
const {intoHiddenEmail} = HelperFunctions;
const {validateEmail} = Validations;

class ForgotPassword extends React.Component {
  constructor(props) {
    super(props);
    const {isLoggedInUser} = props.route?.params ?? {};
    this.state = {
      checkValidation: false,
      email: {
        value: isLoggedInUser ? intoHiddenEmail(props.userEmail) : '',
        empty: !isLoggedInUser,
        isValid: !!isLoggedInUser,
        status: '',
      },
    };
  }

  componentDidMount() {
    this.navigationOptions();
  }

  navigationOptions = () => {
    this.props.navigation.setOptions({
      title: '',
      headerTransparent: true,
      headerLeft: () => (
        <CNavigationBackButton
          title={translate('BACK.BACK_BUTTON_TITLE')}
          backButtonAction={this.goBack}
        />
      ),
    });
  };

  goBack = () => {
    this.props.navigation.pop();
    this.props.addEmail('', '');
  };

  forgotenPassword = () => {
    const {route, userEmail} = this.props;
    const {isLoggedInUser} = route?.params ?? {};
    if (isLoggedInUser) {
      return this.props.forgotPassword({email: userEmail});
    }
    this.setState({
      checkValidation: true,
    });
    let data = {
      email: this.state.email.value,
    };
    if (this.state.email.isValid) {
      this.props.forgotPassword(data);
    }
  };

  updateEmptyStatusOfEmailField = (email) => {
    return email.trim() === '' ? true : false;
  };

  checkIfEmailIsValid = (email) => {
    return validateEmail(email.trim());
  };

  editEmail = (email, status) => {
    this.setState({
      email: {
        value: email,
        empty: this.updateEmptyStatusOfEmailField(email),
        isValid: this.checkIfEmailIsValid(email),
        status: status,
      },
    });
  };

  getEmailErrorMessage = () => {
    const {
      email: {empty, isValid},
    } = this.state;
    const {enterValidEmail, enterEmail} = translate('forgotPassword');

    if (empty) {
      return enterEmail;
    }
    if (!isValid) {
      return enterValidEmail;
    }

    return null;
  };

  render() {
    const {isLoggedInUser} = this.props.route?.params ?? {};

    return (
      <View style={Style.container}>
        <KeyboardAvoidingView
          behavior={Platform.OS === 'ios' ? 'position' : 'height'}
          keyboardVerticalOffset={Platform.OS === 'android' ? 44 : 0}>
          <View style={Style.innerContainer}>
            <View style={Style.appTitleContainer}>
              <CImage type={'local'} imagePath={LOCAL_PATH.SPLASH_ICON} />
            </View>

            <View style={Style.inputContainer}>
              <CLabel style={Style.inputLabel} text={'Email'} />
              <CText
                testID={testIds.forgottenPasswordEmailField}
                accessibilityLabel={testIds.forgottenPasswordEmailField}
                state={this.state.email.status}
                placeHolderText={'Email'}
                value={this.state.email.value}
                editable={!isLoggedInUser}
                error={!this.state.email.isValid && this.state.checkValidation}
                keyboardType={'email-address'}
                onChangeText={(val) => this.editEmail(val, 'Focused')}
                onSubmitEditing={() =>
                  this.editEmail(this.state.email.value, 'Active')
                }
                errorMessage={this.getEmailErrorMessage()}
                autoCapitalize={'none'}
                textInputStyle={isLoggedInUser && Style.blurredTextInput}
              />
            </View>
          </View>
        </KeyboardAvoidingView>
        <View style={Style.buttonsContainer}>
          <CButton
            testID={testIds.sendLinkToEmailToResetPassword}
            accessibilityLabel={testIds.sendLinkToEmailToResetPassword}
            text={'Send'}
            textStyle={Style.whiteText}
            onPress={this.forgotenPassword}
            buttonContainerStyle={Style.blackButton}
            loading={this.props.email.loading}
          />
        </View>
      </View>
    );
  }
}

const mapStateToProps = ({
  forgottenPassword,
  userProfile: {
    userInfo: {email},
  },
}) => {
  return {
    email: forgottenPassword,
    userEmail: email,
  };
};

const mapDispatchToProps = {
  forgotPassword: forgottenPasswordAction,
  addEmail: addEmailAction,
};

export default connect(mapStateToProps, mapDispatchToProps)(ForgotPassword);
