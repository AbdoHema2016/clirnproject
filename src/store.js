import {createStore, applyMiddleware} from 'redux';
import createSagaMiddleware from 'redux-saga';
import {createLogger} from 'redux-logger';
import rootReducer from './reducers';
import {rootSaga} from './sagas';
const sagaMiddleware = createSagaMiddleware();
const isDevelopmentEnv =
  process && process.env && process.env.NODE_ENV === 'development';
const isProductionEnv =
  process && process.env && process.env.NODE_ENV === 'production';
const logStoreInfo = false;
const middlewares = [sagaMiddleware];
function actionLogger({getState}) {
  return (next) => (action) => {
    let allowed = [
      'jumio',
      'notificationsReducer',
      'otp',
      'companies',
      'signIn',
      'updateContactDetailsReducer',
      'userProfile',
      'vaccinations',
    ];
    let state = getState();
    let loggedStates = Object.keys(state)
      .filter((key) => allowed.includes(key))
      .reduce((obj, key) => {
        return {
          ...obj,
          [key]: state[key],
        };
      }, {});
    console.info({actionType: action.type, loggedStates});
    const returnValue = next(action);
    return returnValue;
  };
}
if (isDevelopmentEnv && logStoreInfo) {
  middlewares.push(createLogger());
}
if (isProductionEnv) {
  middlewares.push(actionLogger);
}

const store = createStore(rootReducer, applyMiddleware(...middlewares));
sagaMiddleware.run(rootSaga);
export default store;
