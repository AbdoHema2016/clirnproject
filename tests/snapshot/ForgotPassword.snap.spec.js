import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import store from '../../src/store';
import ForgotPassword from '../../src/containers/onBoarding/forgotPassword/ForgotPassword';

jest.mock('@react-native-firebase/messaging', () => {});
jest.mock('react-native-localize', () => {});

jest.useFakeTimers();

describe('ForgotPassword screen', () => {
  it('Snapshot ForgotPassword', () => {
    const props = {
      navigation: {
        setOptions: jest.fn(),
      },
    };
    const tree = renderer
      .create(
        <Provider store={store}>
          <ForgotPassword {...props} />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
