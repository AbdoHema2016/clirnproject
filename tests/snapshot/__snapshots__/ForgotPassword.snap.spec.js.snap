// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotPassword screen Snapshot ForgotPassword 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F7F7F7",
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
  >
    <View
      style={
        Object {
          "alignItems": "center",
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "marginTop": 302.818,
            "width": "auto",
          }
        }
      >
        <View
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              undefined,
            ]
          }
        >
          <FastImageView
            resizeMode="cover"
            source={
              Object {
                "cache": "immutable",
                "uri": undefined,
              }
            }
            style={
              Object {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "height": "auto",
            "marginTop": 68,
            "width": 726,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#2B342C",
                "fontFamily": "SofiaProRegular",
                "fontSize": 12,
                "height": 16,
                "marginTop": 16,
              },
            ]
          }
        >
          Email
        </Text>
        <View
          style={
            Array [
              Object {
                "height": "auto",
                "width": 724,
              },
              undefined,
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "backgroundColor": "#FFFFFF",
                  "borderRadius": 4,
                  "height": 48,
                  "justifyContent": "center",
                  "marginTop": 4,
                  "width": "100%",
                },
                undefined,
                null,
              ]
            }
          >
            <TextInput
              accessibilityLabel="ForgottenPasswordEmailField"
              accessible={true}
              allowFontScaling={true}
              autoCapitalize="none"
              blurOnSubmit={true}
              editable={true}
              keyboardType="email-address"
              numberOfLines={1}
              onChangeText={[Function]}
              onSubmitEditing={[Function]}
              placeholder="Email"
              placeholderTextColor="#8D8C91"
              rejectResponderTermination={true}
              style={
                Array [
                  Object {
                    "color": "#2B342C",
                    "fontSize": 18,
                    "fontWeight": "500",
                    "lineHeight": 20,
                    "paddingLeft": 19,
                    "paddingRight": 19,
                    "width": "100%",
                  },
                  Object {},
                ]
              }
              testID="ForgottenPasswordEmailField"
              underlineColorAndroid="transparent"
              value=""
            />
          </View>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "bottom": 160.07999999999998,
        "height": "auto",
        "position": "absolute",
        "width": 722,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 4,
            "height": 52,
            "justifyContent": "center",
            "shadowColor": "#DDDDDD",
            "shadowOffset": Object {
              "height": 8,
              "width": 0,
            },
            "shadowOpacity": 0.75,
            "shadowRadius": 4,
          },
          Object {
            "backgroundColor": "#565655",
          },
        ]
      }
    >
      <View
        accessibilityLabel="SendLinkToEmailToResetPassword"
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": undefined,
            "borderRadius": 4,
            "height": "100%",
            "justifyContent": "center",
            "opacity": 1,
            "width": 718,
          }
        }
        testID="SendLinkToEmailToResetPassword"
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "fontFamily": "SofiaProRegular",
                "fontSize": 17,
                "textAlign": "center",
              },
              Object {
                "color": "#FFFFFF",
              },
            ]
          }
        >
          Send
        </Text>
      </View>
    </View>
  </View>
</View>
`;
